pkg/ceph/osd/daemon.go

```diff
@@ -38,6 +38,11 @@ const (
 )
 
 func Run(context *clusterd.Context, agent *OsdAgent) error {
+	if err := setHostName(context); err != nil {
```

Comment:
it actually is using the downward api. The way to access the downward api is by setting an env var. It is set here: https://github.com/travisn/rook/blob/bind-addr/pkg/operator/osd/osd.go#L259